### PR TITLE
type(Table): TypeScript improvement

### PR DESCRIPTION
**Pull request by li-jia-nan**
[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE_CN.md?plain=1)

### 🤔 This is a ...

- [ ] 🆕 New feature
- [ ] 🐞 Bug fix
- [ ] 📝 Site / documentation improvement
- [ ] 📽️ Demo improvement
- [ ] 💄 Component style improvement
- [x] 🤖 TypeScript definition improvement
- [ ] 📦 Bundle size optimization
- [ ] ⚡️ Performance optimization
- [ ] ⭐️ Feature enhancement
- [ ] 🌐 Internationalization
- [ ] 🛠 Refactoring
- [ ] 🎨 Code style optimization
- [ ] ✅ Test Case
- [ ] 🔀 Branch merge
- [ ] ⏩ Workflow
- [ ] ❓ Other (about what?)

### 🔗 Related Issues

> - Describe the source of related requirements, such as links to relevant issue discussions.
> - For example: close #xxxx, fix #xxxx

### 💡 Background and Solution

> - The specific problem to be addressed.
> - List the final API implementation and usage if needed.
> - If there are UI/interaction changes, consider providing screenshots or GIFs.

### 📝 Change Log

> - Describe the impact of the changes on developers, not the solution approach.
> - Reference: https://ant.design/changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | type(Table): TypeScript improvement |
| 🇨🇳 Chinese | 将 Table 中所有的 Record 类型约束为 object，不再定义为 any |

### CODE REVIEW

File: components/table/InternalTable.tsx [hunk 1]
```diff
@@ -74,12 +74,7 @@ interface ChangeEventInfo<RecordType> {
   resetPagination: (current?: number, pageSize?: number) => void;
 }
 
-/** Same as `TableProps` but we need record parent render times */
-export interface InternalTableProps<RecordType> extends TableProps<RecordType> {
-  _renderTimes: number;
-}
-
-export interface TableProps<RecordType = any>
+export interface TableProps<RecordType = AnyObject>
```
Comment (li-jia-nan) [inline components/table/InternalTable.tsx:77]: 这个文件中的 `RecordType` 只能给默认值，不能约束范围，因为对用户暴露了，约束范围会对用户造成 breaking change

File: components/table/interface.ts [hunk 2]
```diff
@@ -66,17 +66,18 @@ export type SorterTooltipProps = TooltipProps & {
...
+export type CompareFn<T = AnyObject> = (a: T, b: T, sortOrder?: SortOrder) => number;
 
 export interface ColumnFilterItem {
   text: React.ReactNode;
   value: React.Key | boolean;
   children?: ColumnFilterItem[];
 }
 
-export interface ColumnTitleProps<RecordType> {
+export interface ColumnTitleProps<RecordType = AnyObject> {
```
Comment (li-jia-nan) [inline components/table/interface.ts:80]: 这个文件中的 `RecordType` 只能给默认值，不能约束范围，理由同上

**Comment by li-jia-nan** [comment]
感觉放在 feature branch 会更好一点？

**Review by afc163 — APPROVED** [review]

**Comment by li-jia-nan** [comment]
@crazyair 大佬表示造成了一些 break change，需要把一部分类型回滚为 any（不过我感觉问题不大，目前没有看到 issue，可以再观察观察）

相关 PR：

- #50372

**Comment by crazyair** [comment]
> @crazyair 大佬表示造成了一些 break change，需要把一部分类型回滚为 any（不过我感觉问题不大，目前没有看到 issue，可以再观察观察）
> 
> 相关 PR：
> 
> * [type: Table generic default any #50372](https://github.com/ant-design/ant-design/pull/50372)

可以到 pro-components 试下

**Pull request merged**